File: Java/src/OOPS/Constructor.java
package OOPS;
public class Constructor {
    public static class Student{
        String name;
        int rno;
        double percentage;
        final String SchName="SMB";
        private static int noOfStudents;
        public Student(String s,int r,double p){      //Parametrized Constructor
            name=s;
            rno=r;
            percentage=p;
            noOfStudents++;
        }
        public Student(Student s){      //copy constructor
            name=s.name;
            rno=s.rno;
            percentage=s.percentage;
        }
    }

    public static void main(String[] args) {
        Student s=new Student("Suman",77,85);
        System.out.println(s.name);
        System.out.println(s.rno);
        System.out.println(s.percentage);
        System.out.println(s.SchName);
        System.out.println(s.noOfStudents);
//        Student n=new Student(s);        //Initializing object of Student class with the help of another object of Student class by using copy constructor
//        System.out.println(n.name);
//        System.out.println(n.rno);
//        System.out.println(n.percentage);
//        System.out.println(n.SchName);

        Student n=new Student("Neha",60,85);
        System.out.println(n.name);
        System.out.println(s.rno);
        System.out.println(s.SchName);
        System.out.println(s.percentage);
        System.out.println(s.noOfStudents);
        System.out.println(Student.noOfStudents);
    }
}

File: c++/OOPS in C++/constructor.cc
#include<iostream>
using namespace std;
class Rectangle{
public:
    int l;
    int b;
    Rectangle(int x, int y){                 // Parameterised constructor
        l=x;
        b=y;
    }
    Rectangle(){                             // Default constructor
        l=0;
        b=0;
    }
    Rectangle(Rectangle &r){                 // Copy constructor
        l=r.l;
        b=r.b;
    }
};
int main(){
    Rectangle r1;
    cout<<r1.l<<" "<<r1.b<<endl;
    Rectangle r2(3,4);
    cout<<r2.l<<" "<<r2.b<<endl;
    Rectangle r3=r2;
    cout<<r3.l<<" "<<r3.b<<endl;
    return 0;
}